File: packages/expo-dev-menu/vendored/react-native-gesture-handler/android/devmenu/com/swmansion/gesturehandler/react/RNGestureHandlerEventDataExtractor.java
package devmenu.com.swmansion.gesturehandler.react;

import com.facebook.react.bridge.WritableMap;
import devmenu.com.swmansion.gesturehandler.GestureHandler;

public interface RNGestureHandlerEventDataExtractor<T extends GestureHandler> {
  void extractEventData(T handler, WritableMap eventData);
}

File: packages/expo-dev-menu/vendored/react-native-gesture-handler/android/devmenu/com/swmansion/gesturehandler/react/RNGestureHandlerEventDataExtractor.kt
package devmenu.com.swmansion.gesturehandler.react

import com.facebook.react.bridge.WritableMap
import devmenu.com.swmansion.gesturehandler.GestureHandler

interface RNGestureHandlerEventDataExtractor<T : GestureHandler<T>> {
  fun extractEventData(handler: T, eventData: WritableMap)
}
